application/src/lib.rs

```diff
@@ -0,0 +1,331 @@
+// Copyright (c) 2022 RBB S.r.l
```

Comment:
Honestly, I don't like the name `application`. The `application`, or similar names, should be reserved for something like the `node` application, which already Aaro pointed out.

The `src` will contain the library components of the mintlayer.
The `node` will contain the executable of the node that imports everything it needs from src.

All the technical tools should remain in this crate, while the simple main and relevant imports should go to the `node` crate. This crate should be renamed to something simple... maybe `subsystem`?

Reply:
Yeah I had this thought too but couldn't think of a name I actually liked

Reply:
My issue with the name `subsystem` is we then have a "subsystem" called `subsystem` so discussing "subsystems" becomes, well as messy as this comment has been.

Reply:
I don't like the name either. I can call it `subsystem` but a prominent part of the crate is a top-level type that manages the subsystems so it would be slightly misleading.

Reply:
Maybe rename application to SubsystemManager? Then calling the crate subsystem wouldn't be so bad.

Reply:
> Maybe rename application to SubsystemManager? Then calling the crate subsystem wouldn't be so bad.

This sounds good to me. I think we can then call it just `Manager` and refer to it by `subsystem::Manager` from other crates.